Userland/Libraries/LibJS/Runtime/DateConstructor.cpp

```diff
@@ -176,25 +176,46 @@ Value DateConstructor::construct(Function&)
         auto milliseconds = static_cast<u16>(tv.tv_usec / 1000);
         return Date::create(global_object(), datetime, milliseconds);
     }
+
     if (vm.argument_count() == 1) {
         auto value = vm.argument(0);
         if (value.is_string())
             value = parse_simplified_iso8601(value.as_string().string());
+
+        if (!value.to_number(global_object()).is_finite_number()) {
+            auto datetime = Core::DateTime::from_timestamp(static_cast<time_t>(0));
+            auto milliseconds = static_cast<u16>(0);
+            return Date::create(global_object(), datetime, milliseconds, true);
+        }
+
         // A timestamp since the epoch, in UTC.
-        // FIXME: This doesn't construct an "Invalid Date" object if the argument is NaN.
-        // FIXME: Date() probably should use a double as internal representation, so that NaN arguments and larger offsets are handled correctly.
-        double value_as_double = value.to_double(global_object());
+        double value_as_double = value.as_double();
```

Comment:
This makes the constructor crash, the above checks the result of `value.to_number()`, that doesn't make `value` a number itself (which is required to call `as_double()` on it)